logging-manager.js

```diff
@@ -41,18 +43,41 @@ const Logger = (module.exports = {
     return this
   },
 
-  checkLogLevel() {
-    fs.readFile('/logging/tracingEndTime', (error, end) => {
-      if (error || !end) {
-        this.logger.level(this.defaultLevel)
-        return
-      }
-      if (parseInt(end) > Date.now()) {
+  async checkLogLevelFile() {
+    try {
+      const end = await fs.promises.readFile('/logging/tracingEndTime')
+      if (!end) throw new Error("No end time found")
+      if (parseInt(end,10) > Date.now()) {
         this.logger.level('trace')
       } else {
         this.logger.level(this.defaultLevel)
       }
-    })
+    } catch (err) {
+      this.logger.level(this.defaultLevel)
+      return
+    }
+  },
+
+async checkLogLevelMetadata() {
+    const options = {
+      headers: {
+        'Metadata-Flavor': 'Google'
+      }
+    }
+    const uri = `http://metadata.google.internal/computeMetadata/v1/project/attributes/${this.loggerName}-setLogLevelEndTime`
+    try {
+      const res = await fetch(uri,options)
+      if (!res.ok) throw new Error("Metadata not okay")
+        const body = await res.text()
+        if (parseInt(body) > Date.now()) {
```

Comment:
```suggestion
        if (parseInt(body, 10) > Date.now()) {
```
For consistency with `checkLogLevelFile`